homeassistant/components/incomfort/water_heater.py

```diff
@@ -0,0 +1,96 @@
+"""Support for an Intergas boiler via an InComfort/Intouch Lan2RF gateway."""
+import asyncio
+import logging
+
+from homeassistant.components.water_heater import WaterHeaterDevice
+from homeassistant.const import TEMP_CELSIUS
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+INCOMFORT_SUPPORT_FLAGS = 0
+
+INCOMFORT_MAX_TEMP = 80.0
+INCOMFORT_MIN_TEMP = 30.0
+
+
+async def async_setup_platform(hass, hass_config, async_add_entities,
+                               discovery_info=None):
+    """Set up an InComfort/Intouch water_heater device."""
+    client = hass.data[DOMAIN]['client']
+    heater = hass.data[DOMAIN]['heater']
+
+    async_add_entities([
+        IncomfortWaterHeater(client, heater)], update_before_add=True)
+
+
+class IncomfortWaterHeater(WaterHeaterDevice):
+    """Representation of an InComfort/Intouch water_heater device."""
+
+    def __init__(self, client, boiler):
+        """Initialize the water_heater device."""
+        self._client = client
+        self._boiler = boiler
+        self._name = 'Boiler'
+
+    @property
+    def name(self):
+        """Return the name of the water_heater device."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        keys = ['nodenr', 'rf_message_rssi', 'rfstatus_cntr', 'room_1',
+                'room_2']
+        state = {k: self._boiler.status[k]
+                 for k in self._boiler.status if k not in keys}
+        return {'status': state}
```

Comment:
Why should we nest the items under the `'status'` key?

Reply:
I can't think of an example with _this_ specific component, but in other cases (e.g. **evohome**), I have had distinct pieces of data, but with the same key, e.g.:
```json
{
  "status_v1": {"temperature": 25.3},
  "status_v2": {"temperature": 32.33}
}
```
Doing it this way let's me (us) extend the device state attributes schema in future _without_ have to rename keys, e.g.:
```json
{
  "temperature": 25.3
}
```
... becomes:
```json
{
  "temperature": 25.3,
  "temperature_v2": 32.33
}

Reply:
Do you think that is probable? We should prefer flat over nested if there's no specific need for the nesting.

Reply:
I have written 3 integrations: **evohome**, **geniushub** and **incomfort**. The first two are both hierarchical, and are both likely to be extended in the future (I have plans for this).

However, this one is flat, and I doubt it will be extended in the future, so will remove the nesting.